src/QsCompiler/QirGeneration/README.md

```diff
@@ -0,0 +1,64 @@
+# QIR Emission - Preview Feature
+
+This file contains directions for using the preview feature integrated into the Q# compiler to emit QIR.
+QIR is a convention for how to represent quantum programs in LLVM. Its specification can be found [here](https://github.com/microsoft/qsharp-language/tree/main/Specifications/QIR#quantum-intermediate-representation-qir). 
+We aim to ultimately move the Q# compiler to be fully LLVM-based. While the emission is supported starting with the March 2021 release, it is as of this time not yet connected to the runtime. The QIR runtime and instructions for how to execute the emitted QIR can be found [here](https://github.com/microsoft/qsharp-runtime/tree/main/src/QirRuntime#the-native-qir-runtime). We are working on a full integration in the future.
+
+## Using the Q# Compiler to Emit QIR
+
+QIR can be emitted for any Q# project as long as its output type is an executable.
+To enable QIR emission, open the project file in a text editor and add the following project property:
+```
+<QirGeneration>true</QirGeneration>
+```
+If the project builds successfully, the .ll file containing QIR can be found in `qir` folder in the project folder. Alternatively, the folder path can be specified via the `QirOutputPath` project property. The project file should look similar to this:
+```
+<Project Sdk="Microsoft.Quantum.Sdk/0.15.2102128704-alpha">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <QirOutputPath>$(MSBuildThisFileDirectory)/qir</QirOutputPath>
+    <QscVerbosity>Detailed</QscVerbosity>
+  </PropertyGroup>
+</Project>
+```
+For more information about project properties and other Sdk capabilities, see [here](https://github.com/microsoft/qsharp-compiler/tree/main/src/QuantumSdk#the-microsoftquantumsdk-nuget-package).
+
+## Limitations
+
+Please be aware that as of this time, it is not possible to both QIR and C#. 
+Open issues related to QIR emission can be found [here](https://github.com/microsoft/qsharp-compiler/issues?q=is%3Aopen+is%3Aissue+label%3A%22area%3A+QIR%22). 
+The emitted QIR does currently not contain any debug information. We are planning to add better support in the future. If you are interested in contributing, please indicate your interest on [this issue](https://github.com/microsoft/qsharp-compiler/issues/637).
+
+### Entry Points
+
+For Q# operation decorated with the `@EntryPoint` attribute, the QIR generation
+will create an additional C-callable wrapper function.
+The name of this wrapper is the same as the full, namespace-qualified name
+of the entry point, with periods replaced by double underscores.
+The entry point wrapper function gets tagged with an custom LLVM "EntryPoint" attribute.
```

Comment:
If the concept of the entry points is introduced in some earlier piece of documentation then placing, in the beginning of this paragraph, a link to that introduction would help. 
Otherwise, despite the fact that this paragraph talks about the entry points, strictly speaking this paragraph still _does not  define_ what the entry point is. In this case you might want to consider something like changing the first sentence to the following:
> _Entry point_ is a Q# operation decorated with the `@EntryPoint` attribute. The Q# _function_ can also be made an entry point (by decorating it with the `@EntryPoint` attribute), but it will make little sense since such a function will not be able to call the _operations_ (see details in \<Functions vs. Operations>).  
For the entry points the Q# compiler will generate an additional LLVM IR wrapper function callable from C code.